api/v1beta1/flowcollector_webhook.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+	"github.com/netobserv/network-observability-operator/api/v1alpha1"
+	utilconversion "github.com/netobserv/network-observability-operator/pkg/conversion"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this v1beta1 FlowCollector to its v1alpha1 equivalent (the conversion Hub)
+// https://book.kubebuilder.io/multiversion-tutorial/conversion.html
+func (r *FlowCollector) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha1.FlowCollector)
+
+	if err := Convert_v1beta1_FlowCollector_To_v1alpha1_FlowCollector(r, dst, nil); err != nil {
+		return fmt.Errorf("copying v1beta1.FlowCollector into v1alpha1.FlowCollector: %w", err)
+	}
+	dst.Status.Conditions = make([]v1.Condition, len(r.Status.Conditions))
+	copy(dst.Status.Conditions, r.Status.Conditions)
+
+	// Manually restore data.
+	restored := &FlowCollector{}
+	if ok, err := utilconversion.UnmarshalData(r, restored); err != nil || !ok {
+		return err
+	}
```

Comment:
Is this the right way around, just thinking that this is convert to v1alpha1, and you're comment is about restoring data?

Normally you would have the annotation stored on the v1alpha1 and you restore when you convert from v1alpha1 to v1beta1, but that seems like it's backwards here

Once you have the `restored` object, you need to then copy the fields from restored to dst that don't exist in the v1alpha1 API right? So `restored` needs to be the same types as the destination, which it isn't right now

Reply:
as discussed over slack we made v1beta1 as the hub and v1alpha1 as the spoke and that now align with the current annotation logic and this is much cleaner now so once we release `v1beta1` all we need is just to make the storage and all code stay as is which is great